Bazel CI not finding test failures related

From https://github.com/cockroachdb/cockroach/pull/84608:

[GitHub CI Run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/7732261?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true)
[Bazel CI Run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/7733392?showRootCauses=false&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true)

GitHub CI reports the following legitimate failures which Bazel CI does not see:

```
=== RUN   TestLint/TestStaticCheck
    lint_test.go:1751:
        pkg/sql/logictest/logic.go:1219:6: func getFreePort is unused (U1000)
    lint_test.go:1751:
        pkg/sql/logictest/logic.go:1240:21: func (*logicTest).newTestServerCluster is unused (U1000)
    lint_test.go:1751:
        pkg/sql/logictest/logic.go:1817:3: this value of bootstrapBinaryPath is never used (SA4006)
    --- FAIL: TestLint/TestStaticCheck (959.91s)
```

Epic CRDB-15060

Jira issue: CRDB-21877

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1233,6 +1233,10 @@ func getFreePort() (int, error) {
 	return port, err
 }
 
+// Prevent a lint failure "this value is never used" in
+// `(*logicTest).setup` when bazel.BuiltWithBazel returns false.
+var _ = ((*logicTest)(nil)).newTestServerCluster
+
 // newTestServerCluster creates a 3-node cluster using the cockroach-go library.
 // bootstrapBinaryPath is given by the config's CockroachGoBootstrapVersion.
 // upgradeBinaryPath is given by the config's CockroachGoUpgradeVersion, or
@@ -1823,6 +1827,11 @@ func (t *logicTest) setup(
 		if err != nil {
 			t.Fatal(err)
 		}
+
+		// Prevent a lint failure "this value is never used" when
+		// bazel.BuiltWithBazel returns false above.
+		_ = bootstrapBinaryPath
+
 		localBinaryPath, found := bazel.FindBinary("pkg/cmd/cockroach-short/cockroach-short_/", "cockroach-short")
 		if !found {
 			t.Fatal(errors.New("cockroach binary not found"))

